### PR TITLE
Remove buck2 CI from pull.yml

### PR DESCRIPTION
**Pull request by kirklandsign**
Keep one model E2E (mv3) from gather_models only for basic validation

We removed all buck2 from docs already

Test: CI

### CODE REVIEW

File: .github/workflows/pull.yml [hunk 1]
```diff
@@ -35,9 +35,6 @@ jobs:
     name: test-setup-linux-gcc
     uses: pytorch/test-infra/.github/workflows/linux_job.yml@main
     strategy:
```
Comment (huydhn) [inline .github/workflows/pull.yml:37]: I think you could just remove this `strategy` section, it does nothing now

File: .github/workflows/pull.yml [hunk 2]
```diff
@@ -411,14 +377,14 @@ jobs:
...
-        BUILD_TOOL=${{ matrix.build-tool }}
+        BUILD_TOOL="cmake"
         MODE=${{ matrix.mode }}
 
         PYTHON_EXECUTABLE=python bash .ci/scripts/setup-qnn-deps.sh
         PYTHON_EXECUTABLE=python bash .ci/scripts/build-qnn-sdk.sh
 
         # Setup executorch
-        PYTHON_EXECUTABLE=python bash .ci/scripts/setup-linux.sh buck2
+        PYTHON_EXECUTABLE=python bash .ci/scripts/setup-linux.sh cmake
```
Comment (huydhn) [inline .github/workflows/pull.yml:387]: Nit: Use `${BUILD_TOOL}` here?

File: .github/workflows/pull.yml [hunk 3]
```diff
@@ -117,11 +105,11 @@ jobs:
         conda activate "${CONDA_ENV}"
 
         DTYPE=${{ matrix.dtype }}
-        BUILD_TOOL=${{ matrix.build-tool }}
+        BUILD_TOOL="cmake"
         MODE=${{ matrix.mode }}
 
         # Setup executorch
-        PYTHON_EXECUTABLE=python bash .ci/scripts/setup-linux.sh buck2
+        PYTHON_EXECUTABLE=python bash .ci/scripts/setup-linux.sh cmake
```
Comment (huydhn) [inline .github/workflows/pull.yml:112]: Nit: Use `${BUILD_TOOL}` here?

**Review by huydhn — APPROVED** [review]
LGTM with some minor comments

File: .github/workflows/pull.yml (hunk 2 above)
Comment (larryliu0820) [reply .github/workflows/pull.yml:387]: Or remove BUILD_TOOL

File: .github/workflows/pull.yml (hunk 1 above)
Comment (kirklandsign) [reply .github/workflows/pull.yml:37]: There is a `fail-fast: false` below

**Comment by kirklandsign** [comment]
Saves >30 minutes CPU time for each pull.yml

**Pull request merged**